R/utilities-sensitivity-analysis.R

```diff
@@ -1,342 +1,343 @@
+#' @param numberOfCoresToUse Number of cores to use on local node.  This parameter should be should be set to 1 when parallelizing over many nodes.
+#' @return Save sensitivity analysis results as CSV in path given by resultsFilePath.
+#' @import ospsuite
+#' @export
+analyzeCoreSensitivity <- function(simulation,
+                                   parametersToPerturb = NULL,
+                                   totalSensitivityThreshold = 1,
+                                   resultsFilePath = paste0(getwd(), "sensitivityAnalysisResults.csv"),
+                                   numberOfCoresToUse = NULL) {
+  sensitivityAnalysis <- SensitivityAnalysis$new(simulation = simulation)
+  sensitivityAnalysis$addParameterPaths(parametersToPerturb)
+
+
+
+  if (is.null(numberOfCoresToUse)) {
+    sensitivityAnalysisRunOptions <- SensitivityAnalysisRunOptions$new(showProgress = FALSE)
+  }
+  else {
+    sensitivityAnalysisRunOptions <- SensitivityAnalysisRunOptions$new(
+      showProgress = FALSE,
+      numberOfCoresToUse = numberOfCoresToUse
+      #The numberOfCoresToUse input in the SensitivityAnalysisRunOptions initializer should not be set to NULL.
+    )
+  }
+  print("Running sensitivity analysis...")
+  sensitivityAnalysisResults <- runSensitivityAnalysis(
+    sensitivityAnalysis = sensitivityAnalysis,
+    sensitivityAnalysisRunOptions = sensitivityAnalysisRunOptions
+  )
+  print("...done")
+  exportSensitivityAnalysisResultsToCSV(results = sensitivityAnalysisResults, resultsFilePath)
+}
+
+
+
+#' @title getPKResultsDataFrame
+#' @description Read PK parameter results into a dataframe and set QuantityPath,Parameter and Unit columns as factors
+#' @param pkParameterResultsFilePath Path to PK parameter results CSV file
+#' @return pkResultsDataFrame, a dataframe storing the contents of the CSV file with path pkParameterResultsFilePath
+#' @import ospsuite
+getPKResultsDataFrame <- function(pkParameterResultsFilePath) {
+  pkResultsDataFrame <- read.csv(pkParameterResultsFilePath, encoding = "UTF-8", check.names = FALSE)
+  colnames(pkResultsDataFrame) <- c("IndividualId", "QuantityPath", "Parameter", "Value", "Unit")
+  pkResultsDataFrame$QuantityPath <- as.factor(pkResultsDataFrame$QuantityPath)
+  pkResultsDataFrame$Parameter <- as.factor(pkResultsDataFrame$Parameter)
+  pkResultsDataFrame$Unit <- as.factor(pkResultsDataFrame$Unit)
+  return(pkResultsDataFrame)
+}
+
+
+
+
+#' @title getQuantileIndividualIds
+#' @description Find IDs of individuals whose PK analysis results closest to quantiles given by vector of quantiles quantileVec
+#' @param pkAnalysisResultsDataframe Dataframe storing the PK analysis results for multiple individuals for a single PK parameter and single output path
+#' @return ids, IDs of individuals whose PK analysis results closest to quantiles given by vector of quantiles quantileVec
+getQuantileIndividualIds <- function(pkAnalysisResultsDataframe, quantileVec = c(0.05, 0.5, 0.95)) {
```

Comment:
Default quantile should be part of the method descriptions

Reply:
removed the default here